<ui version="4.0" >
 <class>ConnectionDialog_BASE</class>
 <widget class="QDialog" name="ConnectionDialog_BASE" >
  <property name="windowModality" >
   <enum>Qt::ApplicationModal</enum>
  </property>
  <property name="geometry" >
   <rect>
    <x>0</x>
    <y>0</y>
    <width>502</width>
    <height>426</height>
   </rect>
  </property>
  <property name="sizePolicy" >
   <sizepolicy vsizetype="Fixed" hsizetype="Preferred" >
    <horstretch>0</horstretch>
    <verstretch>0</verstretch>
   </sizepolicy>
  </property>
  <property name="windowTitle" >
   <string>Open COOL connection</string>
  </property>
  <property name="modal" >
   <bool>true</bool>
  </property>
  <layout class="QGridLayout" >
   <item row="0" column="0" colspan="5" >
    <widget class="QTabWidget" name="tabWidget_ConnectionBuilder" >
     <property name="focusPolicy" >
      <enum>Qt::NoFocus</enum>
     </property>
     <property name="currentIndex" >
      <number>0</number>
     </property>
     <widget class="QWidget" name="ConnectionBuilder" >
      <attribute name="title" >
       <string>Connection Builder</string>
      </attribute>
      <layout class="QVBoxLayout" >
       <item>
        <widget class="QLabel" name="label_11" >
         <property name="text" >
          <string>Select connection string creation method</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QRadioButton" name="radioButton_PredefinedConnections" >
         <property name="text" >
          <string>Predefined connections</string>
         </property>
         <property name="checkable" >
          <bool>false</bool>
         </property>
         <property name="checked" >
          <bool>false</bool>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QRadioButton" name="radioButton_AdhocConnections" >
         <property name="text" >
          <string>Adhoc connections</string>
         </property>
         <property name="checked" >
          <bool>true</bool>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QStackedWidget" name="stackedWidget_ConnectionBuilder" >
         <property name="minimumSize" >
          <size>
           <width>410</width>
           <height>240</height>
          </size>
         </property>
         <property name="frameShape" >
          <enum>QFrame::StyledPanel</enum>
         </property>
         <property name="frameShadow" >
          <enum>QFrame::Sunken</enum>
         </property>
         <property name="currentIndex" >
          <number>1</number>
         </property>
         <widget class="QWidget" name="page_0" >
          <layout class="QVBoxLayout" >
           <item>
            <layout class="QHBoxLayout" >
             <item>
              <widget class="QCheckBox" name="checkBox_UseConnectionAlias" >
               <property name="text" >
                <string>Use connection alias:</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QComboBox" name="comboBox_ConnectionAlias" >
               <property name="enabled" >
                <bool>false</bool>
               </property>
               <property name="sizePolicy" >
                <sizepolicy vsizetype="Fixed" hsizetype="Preferred" >
                 <horstretch>1</horstretch>
                 <verstretch>0</verstretch>
                </sizepolicy>
               </property>
               <property name="editable" >
                <bool>true</bool>
               </property>
              </widget>
             </item>
            </layout>
           </item>
           <item>
            <widget class="QGroupBox" name="groupBox_NotUsingConnectionAlias" >
             <property name="title" >
              <string/>
             </property>
             <property name="flat" >
              <bool>true</bool>
             </property>
             <property name="checkable" >
              <bool>false</bool>
             </property>
             <layout class="QGridLayout" >
              <property name="leftMargin" >
               <number>0</number>
              </property>
              <property name="topMargin" >
               <number>0</number>
              </property>
              <property name="rightMargin" >
               <number>0</number>
              </property>
              <property name="bottomMargin" >
               <number>0</number>
              </property>
              <property name="verticalSpacing" >
               <number>0</number>
              </property>
              <item row="0" column="0" >
               <widget class="QLabel" name="label_9" >
                <property name="text" >
                 <string>Database type:</string>
                </property>
               </widget>
              </item>
              <item row="0" column="1" >
               <widget class="QComboBox" name="comboBox_DBType_2" >
                <property name="sizePolicy" >
                 <sizepolicy vsizetype="Fixed" hsizetype="Preferred" >
                  <horstretch>1</horstretch>
                  <verstretch>0</verstretch>
                 </sizepolicy>
                </property>
                <item>
                 <property name="text" >
                  <string>oracle://</string>
                 </property>
                </item>
                <item>
                 <property name="text" >
                  <string>mysql://</string>
                 </property>
                </item>
                <item>
                 <property name="text" >
                  <string>sqlite_file:</string>
                 </property>
                </item>
               </widget>
              </item>
              <item row="1" column="0" >
               <widget class="QLabel" name="label_wildcard_1" >
                <property name="text" >
                 <string>Server:</string>
                </property>
               </widget>
              </item>
              <item row="1" column="1" >
               <widget class="QComboBox" name="comboBox_Server_2" >
                <property name="sizePolicy" >
                 <sizepolicy vsizetype="Fixed" hsizetype="Preferred" >
                  <horstretch>1</horstretch>
                  <verstretch>0</verstretch>
                 </sizepolicy>
                </property>
                <property name="editable" >
                 <bool>true</bool>
                </property>
               </widget>
              </item>
              <item row="2" column="0" >
               <widget class="QLabel" name="label_wildcard_2" >
                <property name="text" >
                 <string>Schema:</string>
                </property>
               </widget>
              </item>
              <item row="2" column="1" >
               <widget class="QComboBox" name="comboBox_Schema_2" >
                <property name="sizePolicy" >
                 <sizepolicy vsizetype="Fixed" hsizetype="Preferred" >
                  <horstretch>1</horstretch>
                  <verstretch>0</verstretch>
                 </sizepolicy>
                </property>
                <property name="editable" >
                 <bool>true</bool>
                </property>
               </widget>
              </item>
             </layout>
            </widget>
           </item>
           <item>
            <layout class="QHBoxLayout" >
             <item>
              <widget class="QLabel" name="label_7" >
               <property name="text" >
                <string>Role:</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QComboBox" name="comboBox_Role" >
               <property name="sizePolicy" >
                <sizepolicy vsizetype="Fixed" hsizetype="Preferred" >
                 <horstretch>1</horstretch>
                 <verstretch>0</verstretch>
                </sizepolicy>
               </property>
               <property name="editable" >
                <bool>true</bool>
               </property>
              </widget>
             </item>
            </layout>
           </item>
           <item>
            <layout class="QHBoxLayout" >
             <property name="spacing" >
              <number>-1</number>
             </property>
             <item>
              <widget class="QLabel" name="label_10" >
               <property name="text" >
                <string>DB Instance:</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QComboBox" name="comboBox_DBName_2" >
               <property name="sizePolicy" >
                <sizepolicy vsizetype="Fixed" hsizetype="Preferred" >
                 <horstretch>1</horstretch>
                 <verstretch>0</verstretch>
                </sizepolicy>
               </property>
               <property name="acceptDrops" >
                <bool>true</bool>
               </property>
               <property name="editable" >
                <bool>true</bool>
               </property>
              </widget>
             </item>
            </layout>
           </item>
          </layout>
         </widget>
         <widget class="QWidget" name="page_1" >
          <layout class="QGridLayout" >
           <item row="0" column="0" >
            <widget class="QLabel" name="label" >
             <property name="text" >
              <string>Database type:</string>
             </property>
            </widget>
           </item>
           <item row="0" column="1" >
            <widget class="QComboBox" name="comboBox_DBType" >
             <item>
              <property name="text" >
               <string>oracle</string>
              </property>
             </item>
             <item>
              <property name="text" >
               <string>mysql</string>
              </property>
             </item>
             <item>
              <property name="text" >
               <string>sqlite</string>
              </property>
             </item>
            </widget>
           </item>
           <item row="1" column="0" >
            <widget class="QLabel" name="label_wildcard_3" >
             <property name="text" >
              <string>Server:</string>
             </property>
            </widget>
           </item>
           <item row="1" column="1" >
            <widget class="QComboBox" name="comboBox_Server" >
             <property name="editable" >
              <bool>true</bool>
             </property>
            </widget>
           </item>
           <item row="2" column="0" >
            <widget class="QLabel" name="label_wildcard_4" >
             <property name="text" >
              <string>Schema:</string>
             </property>
            </widget>
           </item>
           <item row="2" column="1" >
            <widget class="QComboBox" name="comboBox_Schema" >
             <property name="acceptDrops" >
              <bool>true</bool>
             </property>
             <property name="editable" >
              <bool>true</bool>
             </property>
            </widget>
           </item>
           <item row="3" column="0" >
            <widget class="QLabel" name="label_5" >
             <property name="text" >
              <string>DB Instance:</string>
             </property>
            </widget>
           </item>
           <item row="3" column="1" >
            <widget class="QComboBox" name="comboBox_DBName" >
             <property name="acceptDrops" >
              <bool>true</bool>
             </property>
             <property name="editable" >
              <bool>true</bool>
             </property>
            </widget>
           </item>
           <item row="4" column="0" >
            <widget class="QLabel" name="label_3" >
             <property name="text" >
              <string>Login:</string>
             </property>
            </widget>
           </item>
           <item row="4" column="1" >
            <widget class="QComboBox" name="comboBox_Login" >
             <property name="acceptDrops" >
              <bool>true</bool>
             </property>
             <property name="editable" >
              <bool>true</bool>
             </property>
            </widget>
           </item>
           <item row="5" column="0" >
            <widget class="QLabel" name="label_4" >
             <property name="text" >
              <string>Password:</string>
             </property>
            </widget>
           </item>
           <item row="5" column="1" >
            <widget class="QLineEdit" name="lineEdit_Password" >
             <property name="echoMode" >
              <enum>QLineEdit::Password</enum>
             </property>
            </widget>
           </item>
           <item row="6" column="1" >
            <spacer>
             <property name="orientation" >
              <enum>Qt::Vertical</enum>
             </property>
             <property name="sizeHint" >
              <size>
               <width>20</width>
               <height>91</height>
              </size>
             </property>
            </spacer>
           </item>
          </layout>
         </widget>
        </widget>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="ConnectionHistory" >
      <attribute name="title" >
       <string>Connection History</string>
      </attribute>
      <layout class="QGridLayout" >
       <item row="0" column="0" colspan="3" >
        <widget class="QComboBox" name="comboBox_ConnectionHistory" >
         <property name="focusPolicy" >
          <enum>Qt::TabFocus</enum>
         </property>
        </widget>
       </item>
       <item row="1" column="0" >
        <spacer>
         <property name="orientation" >
          <enum>Qt::Horizontal</enum>
         </property>
         <property name="sizeHint" >
          <size>
           <width>40</width>
           <height>20</height>
          </size>
         </property>
        </spacer>
       </item>
       <item row="1" column="1" >
        <widget class="QPushButton" name="pushButton_RemoveAllHistory" >
         <property name="focusPolicy" >
          <enum>Qt::NoFocus</enum>
         </property>
         <property name="text" >
          <string>Clear All</string>
         </property>
        </widget>
       </item>
       <item row="1" column="2" >
        <widget class="QPushButton" name="pushButton_RemoveHistoryEntry" >
         <property name="focusPolicy" >
          <enum>Qt::NoFocus</enum>
         </property>
         <property name="text" >
          <string>Remove Entry</string>
         </property>
        </widget>
       </item>
       <item row="2" column="1" >
        <spacer>
         <property name="orientation" >
          <enum>Qt::Vertical</enum>
         </property>
         <property name="sizeHint" >
          <size>
           <width>20</width>
           <height>41</height>
          </size>
         </property>
        </spacer>
       </item>
      </layout>
     </widget>
    </widget>
   </item>
   <item row="1" column="0" colspan="3" >
    <spacer>
     <property name="orientation" >
      <enum>Qt::Horizontal</enum>
     </property>
     <property name="sizeHint" >
      <size>
       <width>21</width>
       <height>20</height>
      </size>
     </property>
    </spacer>
   </item>
   <item row="1" column="3" >
    <widget class="QPushButton" name="pushButton_Cancel" >
     <property name="text" >
      <string>Cancel</string>
     </property>
    </widget>
   </item>
   <item row="1" column="4" >
    <widget class="QPushButton" name="pushButton_Connect" >
     <property name="text" >
      <string>Connect</string>
     </property>
     <property name="default" >
      <bool>true</bool>
     </property>
    </widget>
   </item>
  </layout>
 </widget>
 <tabstops>
  <tabstop>radioButton_PredefinedConnections</tabstop>
  <tabstop>radioButton_AdhocConnections</tabstop>
  <tabstop>checkBox_UseConnectionAlias</tabstop>
  <tabstop>comboBox_ConnectionAlias</tabstop>
  <tabstop>comboBox_DBType_2</tabstop>
  <tabstop>comboBox_Server_2</tabstop>
  <tabstop>comboBox_Schema_2</tabstop>
  <tabstop>comboBox_Role</tabstop>
  <tabstop>comboBox_DBName_2</tabstop>
  <tabstop>comboBox_DBType</tabstop>
  <tabstop>comboBox_Server</tabstop>
  <tabstop>comboBox_Schema</tabstop>
  <tabstop>comboBox_DBName</tabstop>
  <tabstop>comboBox_Login</tabstop>
  <tabstop>lineEdit_Password</tabstop>
  <tabstop>comboBox_ConnectionHistory</tabstop>
  <tabstop>pushButton_Connect</tabstop>
  <tabstop>pushButton_Cancel</tabstop>
 </tabstops>
 <resources/>
 <connections>
  <connection>
   <sender>pushButton_Cancel</sender>
   <signal>clicked()</signal>
   <receiver>ConnectionDialog_BASE</receiver>
   <slot>reject()</slot>
   <hints>
    <hint type="sourcelabel" >
     <x>392</x>
     <y>421</y>
    </hint>
    <hint type="destinationlabel" >
     <x>181</x>
     <y>113</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>comboBox_ConnectionHistory</sender>
   <signal>activated(int)</signal>
   <receiver>pushButton_Connect</receiver>
   <slot>setFocus()</slot>
   <hints>
    <hint type="sourcelabel" >
     <x>181</x>
     <y>49</y>
    </hint>
    <hint type="destinationlabel" >
     <x>495</x>
     <y>421</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>comboBox_DBType</sender>
   <signal>activated(int)</signal>
   <receiver>comboBox_Server</receiver>
   <slot>setFocus()</slot>
   <hints>
    <hint type="sourcelabel" >
     <x>149</x>
     <y>131</y>
    </hint>
    <hint type="destinationlabel" >
     <x>149</x>
     <y>161</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>comboBox_Server</sender>
   <signal>activated(int)</signal>
   <receiver>comboBox_Login</receiver>
   <slot>setFocus()</slot>
   <hints>
    <hint type="sourcelabel" >
     <x>149</x>
     <y>161</y>
    </hint>
    <hint type="destinationlabel" >
     <x>149</x>
     <y>251</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>comboBox_Login</sender>
   <signal>activated(int)</signal>
   <receiver>lineEdit_Password</receiver>
   <slot>setFocus()</slot>
   <hints>
    <hint type="sourcelabel" >
     <x>149</x>
     <y>251</y>
    </hint>
    <hint type="destinationlabel" >
     <x>152</x>
     <y>283</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>lineEdit_Password</sender>
   <signal>returnPressed()</signal>
   <receiver>pushButton_Connect</receiver>
   <slot>setFocus()</slot>
   <hints>
    <hint type="sourcelabel" >
     <x>318</x>
     <y>293</y>
    </hint>
    <hint type="destinationlabel" >
     <x>495</x>
     <y>421</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>checkBox_UseConnectionAlias</sender>
   <signal>toggled(bool)</signal>
   <receiver>comboBox_ConnectionAlias</receiver>
   <slot>setEnabled(bool)</slot>
   <hints>
    <hint type="sourcelabel" >
     <x>129</x>
     <y>146</y>
    </hint>
    <hint type="destinationlabel" >
     <x>335</x>
     <y>146</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>checkBox_UseConnectionAlias</sender>
   <signal>toggled(bool)</signal>
   <receiver>groupBox_NotUsingConnectionAlias</receiver>
   <slot>setDisabled(bool)</slot>
   <hints>
    <hint type="sourcelabel" >
     <x>129</x>
     <y>146</y>
    </hint>
    <hint type="destinationlabel" >
     <x>264</x>
     <y>212</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>comboBox_DBName_2</sender>
   <signal>activated(int)</signal>
   <receiver>pushButton_Connect</receiver>
   <slot>setFocus()</slot>
   <hints>
    <hint type="sourcelabel" >
     <x>300</x>
     <y>331</y>
    </hint>
    <hint type="destinationlabel" >
     <x>474</x>
     <y>405</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
